### DISCUSSION TITLE
Agent Performance and Scalability

### DISCUSSION THREAD

**Discussion by Autonomousweb3Dev** [fetchai/uAgents#522] [General]
Hi Fetch Team,

As an active community member, I'm interested in understanding more about the performance and scalability of Fetch agents. I have a few questions that I hope you can shed some light on:

**Concurrency and Load Handling:**

- How many concurrent requests can a Fetch agent handle during peak load?
- Have you conducted any load testing or stress testing to benchmark the maximum capacity?

**Scalability:**

- What strategies are in place to ensure that Fetch agents scale efficiently with an increasing number of users or requests?
- Are there any known limitations or bottlenecks that might affect scalability?

**Latency and Response Time:**

- What is the average response time of Fetch agents under typical usage conditions?
- How does this response time change under high load, and what measures are being taken to minimize latency?

**Resource Utilization:**

- What is the typical CPU and memory usage for Fetch agents during operation?
- Have there been any efforts to optimize resource utilization, and what are the results so far?

I believe these insights would be valuable for the community to understand the robustness and readiness of Fetch agents for real-world applications. Looking forward to your detailed responses!

Thank you for your ongoing efforts and dedication.

**Comment by abhifetch** [comment]
Hi @Autonomousweb3Dev ,

Thanks for starting this discussion and these are really informative queries. We will get back to you with all the possible answers on this soon.

Thanks,
Abhi

**Comment by abhifetch** [comment]
Hi @Autonomousweb3Dev,

These are good questions, but not easy to answer quickly.Thanks a lot for raising the issue and we are indeed doing load testing and stress testing as part of an ongoing investigation into these kinds of performance capabilities. Once concluded, we should be able provide detailed answers to these.

Whenever this is done, the results will be posted on this discussion itself.
